coordinator/src/node/rollover.rs

```diff
@@ -172,10 +141,77 @@ impl Rollover {
 }
 
 impl Node {
+    async fn check_if_eligible_for_rollover(
+        &self,
+        conn: &mut PgConnection,
+        notifier: mpsc::Sender<OrderbookMessage>,
+        trader_id: PublicKey,
+        network: Network,
+    ) -> Result<()> {
+        tracing::debug!(%trader_id, "Checking if the users positions is eligible for rollover");
+        if let Some(position) = positions::Position::get_position_by_trader(
+            conn,
+            trader_id,
+            vec![PositionState::Open, PositionState::Rollover],
+        )? {
+            let channel = self
+                .inner
+                .get_dlc_channel_signed(&position.trader)?
+                .with_context(|| {
+                    format!("Expect signed dlc channel to exist. trader_id={trader_id}")
+                })?;
+
+            let dlc_channel_id = channel.get_dlc_channel_id(0).with_context(|| {
+                format!("Expected to get dlc_channel id. trader_id={trader_id}")
+            })?;
+
+            let channel = self.inner.get_dlc_channel_by_id(&dlc_channel_id)?;
+            let trader_id = channel.get_counter_party_id();
+            let signed_channel = match channel {
+                Channel::Signed(signed_channel) => signed_channel,
+                _ => bail!("Expected channel to be signed. trader_id={}", trader_id),
+            };
+
+            let (retry_rollover, contract_id) = match position.position_state {
+                PositionState::Rollover => self.rollback_channel_if_needed(conn, signed_channel)?,
+                PositionState::Open => (false, signed_channel.get_contract_id()),
+                _ => bail!("Unexpected position state {:?}", position.position_state),
+            };
+
+            if coordinator_commons::is_eligible_for_rollover(OffsetDateTime::now_utc(), network)
+                && !position.is_expired()
```

Comment:
You could do an early return at the beginning of the function and check if the position has expired. If it is, we wouldn't need to do any of this.

Reply:
I could do an early return if the position is `Open` but IMHO that would make it less readable.